### PR TITLE
Feature/sprind 54 - Activity Logs

### PR DESCRIPTION
**Pull request by sksadjad**
added:
- functionality to add ActivityLogs when receiving/sending credentials

**it still needs to implement the fetching and showing the activities

### CODE REVIEW

File: src/machines/funkeC2ShareMachine.tsx [hunk 1]
```diff
@@ -252,45 +258,93 @@ const createFunkeCShareMachine = (opts: FunkeC2ShareMachineOpts): FunkeC2ShareSt
...
+                  sharedCredential.set(credential.hash, credential);
+                });
+              }
+            }
+          }
+
+          sharedCredential.forEach(credential =>
+            store.dispatch<any>(
+              storeActivityLogging({
+                level: LogLevel.TRACE,
```
Comment (nklomp) [inline src/machines/funkeC2ShareMachine.tsx]: Activity should be info level

File: src/machines/funkeC2ShareMachine.tsx [hunk 2]
```diff
@@ -252,45 +258,93 @@ const createFunkeCShareMachine = (opts: FunkeC2ShareMachineOpts): FunkeC2ShareSt
...
+                system: System.OID4VP,
+                subSystemType: SubSystem.OID4VP_OP,
+                initiatorType: InitiatorType.SYSTEM,
```
Comment (nklomp) [inline src/machines/funkeC2ShareMachine.tsx]: Should be user not system

File: src/machines/funkeC2ShareMachine.tsx [hunk 3]
```diff
@@ -252,45 +258,93 @@ const createFunkeCShareMachine = (opts: FunkeC2ShareMachineOpts): FunkeC2ShareSt
...
+                description: 'decline share credential',
```
Comment (nklomp) [inline src/machines/funkeC2ShareMachine.tsx]: Can we make this a bit more of a human readable sentence, as we will show it in the UI

File: src/services/credentialService.ts [hunk 4]
```diff
@@ -63,7 +72,26 @@ export const getVerifiableCredential = async (args: IGetVerifiableCredentialArgs
...
+  return removeCredentialBranding({filter: [{vcHash: hash}]})
+    .then(() => agent.crsDeleteCredential({hash}))
+    .then(deletionResult => {
+      store.dispatch<any>(
+        storeActivityLogging({
+          subSystemType: SubSystem.OID4VP_OP,
+          initiatorType: InitiatorType.SYSTEM,
+          level: LogLevel.TRACE,
+          system: System.CREDENTIALS,
+          description: 'deleteVerifiableCredential function call',
```
Comment (nklomp) [inline src/services/credentialService.ts]: Credential was deleted by the user

File: src/services/machines/funkeC2ShareMachineService.ts [hunk 5]
```diff
@@ -174,6 +181,30 @@ export const siopSendResponse = async (
...
   }
 
+  sharedCredential.forEach(credential =>
+    store.dispatch<any>(
+      storeActivityLogging({
+        level: LogLevel.TRACE,
+        system: System.OID4VP,
+        subSystemType: SubSystem.OID4VP_OP,
+        initiatorType: InitiatorType.SYSTEM,
+        description: 'siopSendResponse function call',
```
Comment (nklomp) [inline src/services/machines/funkeC2ShareMachineService.ts]: That would be an audit event not an activity

File: src/services/machines/funkeC2ShareMachineService.ts [hunk 6]
```diff
@@ -240,9 +270,33 @@ export const storePIDCredentials = async (context: Pick<FunkeC2ShareMachineConte
...
       opts: {hasher: generateDigest},
     });
+
     if (!parentId) {
       parentId = digitalCredential.id;
+      parentCredentialHash = digitalCredential.hash;
     }
+
+    store.dispatch<any>(
+      storeActivityLogging({
```
Comment (nklomp) [inline src/services/machines/funkeC2ShareMachineService.ts]: That would be an audit event not an activity

File: src/services/machines/funkeC2ShareMachineService.ts [hunk 7]
```diff
@@ -300,3 +354,30 @@ const determineCorrelationId = async (uri: URL | undefined, verifiedAuthorizatio
...
   throw new Error("Can't determine correlationId from request");
 };
+
+const deletePIDCredentials = async (): Promise<void> => {
+  const deleteCredentials = (await getVerifiableCredentialsFromStorage({regulationTypes: [RegulationType.PID], parentsOnly: false})).map(
+    credential => {
+      store.dispatch<any>(deleteVerifiableCredential(credential.hash)).then(() =>
+        store.dispatch<any>(
+          storeActivityLogging({
+            level: LogLevel.TRACE,
```
Comment (nklomp) [inline src/services/machines/funkeC2ShareMachineService.ts]: INFO

File: src/services/machines/funkeC2ShareMachineService.ts [hunk 8]
```diff
@@ -300,3 +354,30 @@ const determineCorrelationId = async (uri: URL | undefined, verifiedAuthorizatio
...
+            system: System.CREDENTIALS,
+            subSystemType: SubSystem.OID4VP_OP,
+            initiatorType: InitiatorType.SYSTEM,
+            description: 'deletePIDCredentials function call',
```
Comment (nklomp) [inline src/services/machines/funkeC2ShareMachineService.ts]: Credential was deleted by user

File: src/services/machines/getPIDCredentialMachineService.ts [hunk 9]
```diff
@@ -63,12 +58,62 @@ export const storePIDCredentials = async (context: Pick<GetPIDCredentialsMachine
...
       opts: {hasher: generateDigest},
     });
+
     if (!parentId) {
       parentId = digitalCredential.id;
+      parentCredentialHash = digitalCredential.hash;
     }
+
+    store.dispatch<any>(
+      storeActivityLogging({
```
Comment (nklomp) [inline src/services/machines/getPIDCredentialMachineService.ts]: Audit event not activity

File: src/services/machines/getPIDCredentialMachineService.ts [hunk 10]
```diff
@@ -63,12 +58,62 @@ export const storePIDCredentials = async (context: Pick<GetPIDCredentialsMachine
...
+  const deleteCredentials = (await getVerifiableCredentialsFromStorage({regulationTypes: [RegulationType.PID], parentsOnly: false})).map(
+    credential => {
+      store.dispatch<any>(deleteVerifiableCredential(credential.hash)).then(() =>
+        store.dispatch<any>(
+          storeActivityLogging({
+            level: LogLevel.TRACE,
+            system: System.CREDENTIALS,
+            subSystemType: SubSystem.OID4VP_OP,
+            initiatorType: InitiatorType.SYSTEM,
+            description: 'deletePIDCredentials function call',
```
Comment (nklomp) [inline src/services/machines/getPIDCredentialMachineService.ts]: Credential was deleted by user

File: src/services/machines/onboardingMachineService.ts [hunk 11]
```diff
@@ -66,10 +66,34 @@ export const storePIDCredentials = async (context: Pick<OnboardingMachineContext
...
+    storeCredentials.push(digitalCredential);
+
     if (!parentId) {
       parentId = digitalCredential.id;
+      parentCredentialHash = digitalCredential.hash;
     }
-    storeCredentials.push(digitalCredential);
+
+    store.dispatch<any>(
+      storeActivityLogging({
```
Comment (nklomp) [inline src/services/machines/onboardingMachineService.ts]: Audit instead of activity

**Review by nklomp — CHANGES_REQUESTED** [review]
A lot of the logs are audit events not activities. Activities are only there for receiving a credential and sharing them. They are only the user initiated actions. So OK on receipt, or active decline on receipt/share. 

Activities should be at level INFO and should have human readable strings (in the future i18n labels, but given the time we have for now just plain English texts).

The other `activities` like related to storage, are Audit events

File: src/machines/funkeC2ShareMachine.tsx (hunk 1 above)
Comment (BtencateSphereon) [reply src/machines/funkeC2ShareMachine.tsx]: changed

File: src/machines/funkeC2ShareMachine.tsx (hunk 2 above)
Comment (BtencateSphereon) [reply src/machines/funkeC2ShareMachine.tsx]: changed

File: src/machines/funkeC2ShareMachine.tsx (hunk 3 above)
Comment (BtencateSphereon) [reply src/machines/funkeC2ShareMachine.tsx]: changed

File: src/services/credentialService.ts (hunk 4 above)
Comment (BtencateSphereon) [reply src/services/credentialService.ts]: changed

File: src/services/machines/funkeC2ShareMachineService.ts (hunk 6 above)
Comment (BtencateSphereon) [reply src/services/machines/funkeC2ShareMachineService.ts]: changed

File: src/services/machines/funkeC2ShareMachineService.ts (hunk 7 above)
Comment (BtencateSphereon) [reply src/services/machines/funkeC2ShareMachineService.ts]: changed

File: src/services/machines/funkeC2ShareMachineService.ts (hunk 8 above)
Comment (BtencateSphereon) [reply src/services/machines/funkeC2ShareMachineService.ts]: changed

File: src/services/machines/getPIDCredentialMachineService.ts (hunk 9 above)
Comment (BtencateSphereon) [reply src/services/machines/getPIDCredentialMachineService.ts]: changed

File: src/services/machines/getPIDCredentialMachineService.ts (hunk 10 above)
Comment (BtencateSphereon) [reply src/services/machines/getPIDCredentialMachineService.ts]: changed

File: src/services/machines/onboardingMachineService.ts (hunk 11 above)
Comment (BtencateSphereon) [reply src/services/machines/onboardingMachineService.ts]: changed

File: src/services/machines/funkeC2ShareMachineService.ts (hunk 5 above)
Comment (BtencateSphereon) [reply src/services/machines/funkeC2ShareMachineService.ts]: "Activities are only there for receiving a credential and sharing them"

this is being used to determine which credential(s) we are sharing. So for now would that be ok? I updated the description to make it human readable

**Pull request merged**